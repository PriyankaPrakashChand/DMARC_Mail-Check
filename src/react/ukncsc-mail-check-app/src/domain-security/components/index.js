export { default as AggregateReportChart } from './AggregateReportChart';
export {
  default as AggregateReportChartLegend,
} from './AggregateReportChartLegend';
export {
  default as AggregateReportDateSelector,
} from './AggregateReportDateSelector';
export { default as AggregateReportSummary } from './AggregateReportSummary';
export {
  default as DomainSecurityDetailsMessages,
} from './DomainSecurityDetailsMessages';
export { default as DomainSecurityDetailsMx } from './DomainSecurityDetailsMx';
export {
  default as DomainSecurityDetailsTxt,
} from './DomainSecurityDetailsTxt';
export { default as DomainSecurityPill } from './DomainSecurityPill';
export { default as DomainSecurityRecord } from './DomainSecurityRecord';
export {
  default as DomainSecurityRecordExplanation,
} from './DomainSecurityRecordExplanation';
export { default as DomainSecuritySummary } from './DomainSecuritySummary';
export {
  default as DomainSecuritySummaryMessages,
} from './DomainSecuritySummaryMessages';
export { default as DomainSecuritySummaryMx } from './DomainSecuritySummaryMx';
export {
  default as DomainSecuritySummarySubdomains,
} from './DomainSecuritySummarySubdomains';
export {
  default as DomainSecuritySummaryTxt,
} from './DomainSecuritySummaryTxt';
export { default as DomainSecurityTitle } from './DomainSecurityTitle';
