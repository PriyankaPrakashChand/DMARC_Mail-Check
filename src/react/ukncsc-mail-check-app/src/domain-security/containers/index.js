export {
  default as AggregateReportSummaryContainer,
} from './AggregateReportSummaryContainer';
export {
  default as DomainSecurityAggregateExport,
} from './DomainSecurityAggregateExport';
export {
  default as DomainSecurityDetailsCertificates,
} from './DomainSecurityDetailsCertificates';
export {
  default as DomainSecurityDetailsDkimContainer,
} from './DomainSecurityDetailsDkimContainer';
export {
  default as DomainSecurityDetailsDmarc,
} from './DomainSecurityDetailsDmarc';
export {
  default as DomainSecurityDetailsSpf,
} from './DomainSecurityDetailsSpf';
export {
  default as DomainSecurityDetailsTls,
} from './DomainSecurityDetailsTls';
export {
  default as DomainSecuritySummaryCertificatesContainer,
} from './DomainSecuritySummaryCertificatesContainer';
export {
  default as DomainSecuritySummaryContainer,
} from './DomainSecuritySummaryContainer';
export {
  default as DomainSecuritySummaryDkimContainer,
} from './DomainSecuritySummaryDkimContainer';
export {
  default as DomainSecuritySummaryDmarcContainer,
} from './DomainSecuritySummaryDmarcContainer';
export {
  default as DomainSecuritySummarySpfContainer,
} from './DomainSecuritySummarySpfContainer';
export {
  default as DomainSecuritySummarySubdomainsContainer,
} from './DomainSecuritySummarySubdomainsContainer';
export {
  default as DomainSecuritySummaryTlsContainer,
} from './DomainSecuritySummaryTlsContainer';
export { default as DomainSecurityTlsAdvice } from './DomainSecurityTlsAdvice';
