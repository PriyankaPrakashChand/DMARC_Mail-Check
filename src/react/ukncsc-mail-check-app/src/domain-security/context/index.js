export {
  default as DomainSecurityLocationContext,
} from './DomainSecurityLocationContext';
