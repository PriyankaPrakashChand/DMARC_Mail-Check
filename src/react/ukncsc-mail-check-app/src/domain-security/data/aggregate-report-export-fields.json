{
  "aggregateReportExportFields": [
    { "title": "header_from", "description": "'From' mail header domain" },
    {
      "title": "source_ip",
      "description": "IP address of the server sending the email"
    },
    { "title": "PTR", "description": "Reverse DNS lookup of the source_ip" },
    { "title": "count", "description": "Number of emails" },
    { "title": "spf", "description": "Whether SPF alignment was pass or fail" },
    {
      "title": "dkim",
      "description": "Whether DKIM alignment was pass or fail"
    },
    {
      "title": "disposition",
      "description":
        "What the receiving server did with the emails (none=delivered, quarrantine=spam folder, reject=not delivered)"
    },
    {
      "title": "org_name",
      "description":
        "Description of the organisation which sent the report this data is from"
    },
    { "title": "effective_date", "description": "Date the report refers to" }
  ]
}
