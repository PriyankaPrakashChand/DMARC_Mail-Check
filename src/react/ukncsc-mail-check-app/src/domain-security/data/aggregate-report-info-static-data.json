{
  "graphDescriptions": [
    {
      "key": "fullyTrusted",
      "title": "Fully trusted",
      "description": "Emails delivered with both successful SPF and DKIM.",
      "color": "#c6cd80",
      "stroke": "#1a1e04"
    },
    {
      "key": "partiallyTrusted",
      "title": "Partially trusted",
      "description": "Emails delivered with either successful SPF or DKIM.",
      "color": "#98f0e9",
      "stroke": "#0b2321"
    },
    {
      "key": "untrusted",
      "title": "Untrusted",
      "description": "Emails delivered where SPF and DKIM both failed.",
      "color": "#9fd0fb",
      "stroke": "#071f3d"
    },
    {
      "key": "quarantined",
      "title": "Quarantined",
      "description":
        "Emails sent to spam folder because both SPF and DKIM failed.",
      "color": "#f5c05e",
      "stroke": "#221803"
    },
    {
      "key": "rejected",
      "title": "Rejected",
      "description":
        "Emails not delivered to the user because both SPF and DKIM failed.",
      "color": "#f0a1f1",
      "stroke": "#3a0936"
    }
  ]
}
