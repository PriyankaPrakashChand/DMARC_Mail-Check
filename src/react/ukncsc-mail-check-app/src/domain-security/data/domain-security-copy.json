{
  "dkimDescription":
    "DomainKeys Identified Mail (DKIM) is an email authentication method designed to detect email spoofing. It allows the receiver to check that an email claimed to have come from a specific domain was indeed authorized by the owner of that domain.",
  "dmarcDescription":
    "Domain-based Message Authentication, Reporting & Conformance (DMARC) is an email authentication, policy, and reporting protocol that ensures that legitimate email is properly authenticating against established DKIM and SPF standards.",
  "spfDescription":
    "Sender Policy Framework (SPF) is an email-validation system designed to detect email spoofing by providing a mechanism to allow receiving mail exchangers to check that incoming mail from a domain comes from a host authorised by that domain's administrators.",
  "tlsDescription":
    "Transport Layer Security (TLS) is an encryption protocol that provides communication security between client/server applications that communicate with each other over the internet.",
  "certsDescription":
    "Certificates are electronic documents used to prove the ownership of a digital entity such as a domain name. They are digitally signed by a trusted third-party and allow identity verification when setting up TLS connections.",
  "emailSecurityDescription":
    "Total emails from the selected date range broken down into those that are Fully Trusted, Partially trusted, Untrusted, Quarantined and Rejected."
}
