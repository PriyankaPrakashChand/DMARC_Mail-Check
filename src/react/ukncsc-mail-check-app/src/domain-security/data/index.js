export * from './domain-security-copy';
export * from './aggregate-report-info-static-data';
export * from './aggregate-report-export-fields';
