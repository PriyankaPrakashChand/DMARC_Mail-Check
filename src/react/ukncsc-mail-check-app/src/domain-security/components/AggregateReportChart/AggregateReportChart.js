import React, { Component } from 'react';
import moment from 'moment';
import reduce from 'lodash/reduce';
import numeral from 'numeral';
import {
  VictoryArea,
  VictoryAxis,
  VictoryStack,
  VictoryTooltip,
  VictoryVoronoiContainer,
} from 'victory';
import { IeFriendlyVictoryChart } from 'common/components';

const pointsReducer = (prevAccumulator, previousValues, x) => (
  accumulator,
  value
) => ({
  ...accumulator,
  [value]: [...(prevAccumulator[value] || []), { x, y: previousValues[value] }],
});

const dataReducer = (accumulator, value, key) => ({
  ...accumulator,
  ...reduce(Object.keys(value), pointsReducer(accumulator, value, key), {}),
});

const CustomTooltip = ({ descriptions, text, date, datum, ...props }) => (
  <VictoryTooltip
    {...props}
    text={[
      date(datum.x),
      ...descriptions.map(({ title }, i) => `${title}: ${text[i]}`).reverse(),
    ]}
  />
);

const tickLabels = {
  fontSize: 12,
  fontWeight: 500,
  fontFamily: 'Helvetica Neue',
};

export default class AggregateReportChart extends Component {
  state = {
    chartData: {},
    tickValues: [],
  };

  static getDerivedStateFromProps = ({ data }) => ({
    chartData: reduce(data, dataReducer, {}),
    tickValues: Object.keys(data),
  });

  render() {
    const { descriptions } = this.props;
    const { chartData, tickValues } = this.state;

    return (
      <IeFriendlyVictoryChart
        width={700}
        height={400}
        containerComponent={
          <VictoryVoronoiContainer
            voronoiDimension="x"
            labels={d => `${d.y}`}
            labelComponent={
              <CustomTooltip
                descriptions={descriptions}
                date={x => moment(x).format('Do MMMM YYYY')}
              />
            }
          />
        }
      >
        <VictoryAxis
          style={{
            tickLabels,
            axis: { strokeWidth: 0 },
            grid: { stroke: '#c1c2c4', strokeWidth: 1 },
          }}
          tickFormat={t => numeral(t).format('0.0a')}
          fixLabelOverlap
          dependentAxis
        />
        <VictoryAxis
          style={{
            tickLabels,
            axis: { stroke: '#c1c2c4' },
            grid: { strokeWidth: 0 },
          }}
          tickValues={tickValues}
          tickFormat={t => moment(t).format('Do MMM YY')}
          fixLabelOverlap
        />
        <VictoryStack>
          {descriptions.map(({ key, color, stroke }) => (
            <VictoryArea
              data={chartData[key]}
              key={key}
              style={{
                data: { fill: color, stroke, strokeWidth: 2 },
              }}
            />
          ))}
        </VictoryStack>
      </IeFriendlyVictoryChart>
    );
  }
}
