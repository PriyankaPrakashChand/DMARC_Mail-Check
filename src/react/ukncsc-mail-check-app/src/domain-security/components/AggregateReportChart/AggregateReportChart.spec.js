import React from 'react';
import { render } from 'react-testing-library';
import { graphDescriptions } from 'domain-security/data';
import AggregateReportChart from './AggregateReportChart';

describe('AggregateReportChart', () => {
  let container;

  describe('when getting an aggregate report chart', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportChart
          descriptions={graphDescriptions}
          data={{
            '2017-12-10T00:00:00': {
              fullyTrusted: 1,
              partiallyTrusted: 2,
              untrusted: 3,
              quarantined: 4,
              rejected: 5,
            },
            '2017-12-11T00:00:00': {
              fullyTrusted: 6,
              partiallyTrusted: 7,
              untrusted: 8,
              quarantined: 9,
              rejected: 10,
            },
            '2017-12-12T00:00:00': {
              fullyTrusted: 11,
              partiallyTrusted: 12,
              untrusted: 13,
              quarantined: 14,
              rejected: 15,
            },
          }}
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
