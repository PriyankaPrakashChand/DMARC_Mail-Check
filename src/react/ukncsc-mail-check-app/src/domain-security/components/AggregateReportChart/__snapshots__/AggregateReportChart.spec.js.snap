// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportChart when getting an aggregate report chart it should match the snapshot 1`] = `
<div>
  <div
    style="position: relative; height: 0px; width: 100%; padding: 0px 0px 57.14285714285714% 0px;"
  >
    <div
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; pointer-events: none;"
      class="VictoryContainer"
    >
      <svg
        width="700"
        height="400"
        role="img"
        aria-labelledby="victory-container-1-title victory-container-1-desc"
        viewBox="0 0 700 400"
        style="pointer-events: all; width: 100%; height: 100%;"
      >
        <g
          role="presentation"
        >
          <line
            x1="50"
            x2="50"
            y1="50"
            y2="350"
            style="stroke: #252525; stroke-width: 0; fill: transparent; stroke-linecap: round; stroke-linejoin: round;"
            role="presentation"
            shape-rendering="auto"
            vector-effect="non-scaling-stroke"
          />
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="303.84615384615387"
              y2="303.84615384615387"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="303.84615384615387"
              y2="303.84615384615387"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="303.84615384615387"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                10.0
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="257.6923076923077"
              y2="257.6923076923077"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="257.6923076923077"
              y2="257.6923076923077"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="257.6923076923077"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                20.0
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="211.53846153846152"
              y2="211.53846153846152"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="211.53846153846152"
              y2="211.53846153846152"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="211.53846153846152"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                30.0
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="165.3846153846154"
              y2="165.3846153846154"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="165.3846153846154"
              y2="165.3846153846154"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="165.3846153846154"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                40.0
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="119.23076923076923"
              y2="119.23076923076923"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="119.23076923076923"
              y2="119.23076923076923"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="119.23076923076923"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                50.0
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="650"
              y1="73.07692307692304"
              y2="73.07692307692304"
              style="stroke: #c1c2c4; stroke-width: 1; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="49"
              y1="73.07692307692304"
              y2="73.07692307692304"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="39"
              dx="0"
              y="73.07692307692304"
              dy="4.26"
            >
              <tspan
                x="39"
                dx="0"
                text-anchor="end"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                60.0
              </tspan>
            </text>
          </g>
        </g>
        <g
          role="presentation"
        >
          <line
            x1="50"
            x2="650"
            y1="350"
            y2="350"
            style="stroke: #c1c2c4; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
            role="presentation"
            shape-rendering="auto"
            vector-effect="non-scaling-stroke"
          />
          <g
            role="presentation"
          >
            <line
              x1="50"
              x2="50"
              y1="350"
              y2="50"
              style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="50"
              x2="50"
              y1="350"
              y2="351"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="50"
              dx="0"
              y="361"
              dy="10.26"
            >
              <tspan
                x="50"
                dx="0"
                text-anchor="middle"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                10th Dec 17
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="350"
              x2="350"
              y1="350"
              y2="50"
              style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="350"
              x2="350"
              y1="350"
              y2="351"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="350"
              dx="0"
              y="361"
              dy="10.26"
            >
              <tspan
                x="350"
                dx="0"
                text-anchor="middle"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                11th Dec 17
              </tspan>
            </text>
          </g>
          <g
            role="presentation"
          >
            <line
              x1="650"
              x2="650"
              y1="350"
              y2="50"
              style="stroke: none; stroke-width: 0; fill: none; pointer-events: painted;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <line
              x1="650"
              x2="650"
              y1="350"
              y2="351"
              style="stroke: transparent; fill: transparent; size: 1px;"
              role="presentation"
              shape-rendering="auto"
              vector-effect="non-scaling-stroke"
            />
            <text
              x="650"
              dx="0"
              y="361"
              dy="10.26"
            >
              <tspan
                x="650"
                dx="0"
                text-anchor="middle"
                style="font-size: 12px; font-weight: 500; font-family: Helvetica Neue; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
              >
                12th Dec 17
              </tspan>
            </text>
          </g>
        </g>
        <g>
          <g
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
          >
            <defs>
              <clippath
                id="victory-clip-2"
              >
                <rect
                  x="50"
                  y="50"
                  width="600"
                  height="300"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#victory-clip-2)"
            >
              <g>
                <path
                  d="M50,345.38461538461536L350,322.3076923076923L650,299.2307692307692L650,350L350,350L50,350Z"
                  style="fill: #c6cd80; stroke: none; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
                <path
                  d="M50,345.38461538461536L350,322.3076923076923L650,299.2307692307692"
                  style="fill: none; stroke: #1a1e04; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
              </g>
            </g>
          </g>
          <g
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
          >
            <defs>
              <clippath
                id="victory-clip-3"
              >
                <rect
                  x="50"
                  y="50"
                  width="600"
                  height="300"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#victory-clip-3)"
            >
              <g>
                <path
                  d="M50,336.15384615384613L350,290L650,243.84615384615384L650,299.2307692307692L350,322.3076923076923L50,345.38461538461536Z"
                  style="fill: #98f0e9; stroke: none; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
                <path
                  d="M50,336.15384615384613L350,290L650,243.84615384615384"
                  style="fill: none; stroke: #0b2321; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
              </g>
            </g>
          </g>
          <g
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
          >
            <defs>
              <clippath
                id="victory-clip-4"
              >
                <rect
                  x="50"
                  y="50"
                  width="600"
                  height="300"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#victory-clip-4)"
            >
              <g>
                <path
                  d="M50,322.3076923076923L350,253.07692307692307L650,183.84615384615384L650,243.84615384615384L350,290L50,336.15384615384613Z"
                  style="fill: #9fd0fb; stroke: none; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
                <path
                  d="M50,322.3076923076923L350,253.07692307692307L650,183.84615384615384"
                  style="fill: none; stroke: #071f3d; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
              </g>
            </g>
          </g>
          <g
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
          >
            <defs>
              <clippath
                id="victory-clip-5"
              >
                <rect
                  x="50"
                  y="50"
                  width="600"
                  height="300"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#victory-clip-5)"
            >
              <g>
                <path
                  d="M50,303.84615384615387L350,211.53846153846152L650,119.23076923076923L650,183.84615384615384L350,253.07692307692307L50,322.3076923076923Z"
                  style="fill: #f5c05e; stroke: none; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
                <path
                  d="M50,303.84615384615387L350,211.53846153846152L650,119.23076923076923"
                  style="fill: none; stroke: #221803; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
              </g>
            </g>
          </g>
          <g
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px;"
          >
            <defs>
              <clippath
                id="victory-clip-6"
              >
                <rect
                  x="50"
                  y="50"
                  width="600"
                  height="300"
                  role="presentation"
                  shape-rendering="auto"
                  vector-effect="non-scaling-stroke"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#victory-clip-6)"
            >
              <g>
                <path
                  d="M50,280.7692307692308L350,165.3846153846154L650,50L650,119.23076923076923L350,211.53846153846152L50,303.84615384615387Z"
                  style="fill: #f0a1f1; stroke: none; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
                <path
                  d="M50,280.7692307692308L350,165.3846153846154L650,50"
                  style="fill: none; stroke: #3a0936; stroke-width: 2;"
                  role="presentation"
                  shape-rendering="auto"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
      <div
        style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
      >
        <svg
          width="700"
          height="400"
          viewBox="0 0 700 400"
          style="overflow: visible; width: 100%; height: 100%;"
        />
      </div>
    </div>
  </div>
</div>
`;
